dapp/src/components/buySell/BuySellWidget.js

```diff
@@ -242,20 +242,30 @@ const BuySellWidget = ({
         vaultContract.autoAllocateThreshold(),
         vaultContract.rebaseThreshold(),
       ])
-
-      if (totalMintAmount.gte(allocateThreshold.mul(96).div(100))) {
-        // Define gas limit only when the amount is over threshold
-        gasLimit = gasLimits.MINT_ALLOCATE_GAS_LIMIT
-      } else if (totalMintAmount.gte(rebaseThreshold.mul(96).div(100))) {
-        gasLimit = gasLimits.MINT_REBASE_GAS_LIMIT
+      const involvingMultipleCoins = mintAddresses.length > 1
+      const aboveAllocateThreshold = totalMintAmount.gte(
```

Comment:
super nit: maybe add a comment as to why we lower the threshold to 96% so that we remember why next time we read this code.

Reply:
Definitely 👍 I inherited that part but I'll add something in.